File: app/src/main/java/net/bloople/manga/MangaApplication.java
package net.bloople.manga;

import android.app.Application;
import android.net.Uri;

import java.util.HashMap;

public class MangaApplication extends Application {
    public static HashMap<Long, Book> allBooks;

    public static void ensureAllBooks(BooksLoadedListener listener) {
        if(allBooks != null) {
            listener.onBooksLoaded();
        }
        else {
            LoadBooksTask loader = new LoadBooksTask(listener);
            loader.execute();
        }
    }

    public static Uri root() {
        return Uri.parse("http://192.168.1.100:82/Manga-OG/.mangos/");
    }
}

File: app/src/main/java/net/bloople/manga/MangaApplication.kt
package net.bloople.manga

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import coil3.ImageLoader
import coil3.PlatformContext
import coil3.SingletonImageLoader
import coil3.network.okhttp.OkHttpNetworkFetcherFactory
import okhttp3.OkHttpClient

class MangaApplication : Application(), SingletonImageLoader.Factory {
    override fun onCreate() {
        super.onCreate()
        context = applicationContext
    }

    override fun newImageLoader(context: PlatformContext): ImageLoader {
        return ImageLoader.Builder(context)
            .components {
                add(OkHttpNetworkFetcherFactory({ okHttpClient }))
            }
            .build()
    }

    companion object {
        @SuppressLint("StaticFieldLeak")
        lateinit var context: Context
            private set

        val okHttpClient = OkHttpClient.Builder().addInterceptor(HttpAuthInterceptor()).build()
    }
}